File: backend/src/main/java/com/interview/platform/mod19_recommendation_engine/dto/RoadmapResponseDto.java
package com.interview.platform.mod19_recommendation_engine.dto;

import java.util.List;

public class RoadmapResponseDto {
    private String title;
    private List<RoadmapMilestoneDto> items;

    public RoadmapResponseDto(String title, List<RoadmapMilestoneDto> items) {
        this.title = title;
        this.items = items;
    }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    public List<RoadmapMilestoneDto> getItems() { return items; }
    public void setItems(List<RoadmapMilestoneDto> items) { this.items = items; }
}

File: android-app/app/src/main/java/com/interview/platform/data/remote/dto/mod19_recommendation_engine/RoadmapResponseDto.kt
package com.interview.platform.data.remote.dto.mod19_recommendation_engine

data class RoadmapResponseDto(
    val title: String,
    val items: List<RoadmapMilestoneDto>
)

data class RoadmapMilestoneDto(
    val title: String,
    val description: String,
    val status: String,
    val progress: Float,
    val topics: List<RoadmapTopicDto> = emptyList()
)

data class RoadmapTopicDto(
    val id: String,
    val name: String,
    val isCompleted: Boolean
)
